### PR TITLE
"Fix Bug 2001169 - Audit event 'ACCESS_SESSION_ESTABLISH' is not gener…

### PR DESCRIPTION
**Pull request by c-dorney**
…ating for PKI instances acting as Server [10.2.1] (#3745) (#3763)

This fix allows us to actually see ssl connection events in the audit log from the pki /server perspective.
This fill will also require support bug fixes for both jss and tomcatjss.

Added fix for stray alerts showing up after a server is going down.

Sample audit log messages:

0.https-jsse-nio-18443-exec-6 - [29/Sep/2021:21:09:31 EDT] [14] [6] [AuditEvent=ACCESS_SESSION_ESTABLISH][ClientIP=--][ServerIP=--][SubjectID=CN=PKI Administrator,E=example@testdomain.com,OU=rhcs94-CA-cfu_rsa-nocp11,O=Example-rhcs94-CA_cfu-rsa][Outcome=Success] access session establish success
0.https-jsse-nio-18443-exec-16 - [29/Sep/2021:21:09:32 EDT] [14] [6] [AuditEvent=AUTHZ][SubjectID=$NonRoleUser$][Outcome=Success][aclResource=certServer.ee.profiles][Op=list] authorization success

0.https-jsse-nio-18443-exec-16 - [29/Sep/2021:21:11:34 EDT] [14] [6] [AuditEvent=ACCESS_SESSION_TERMINATED][ClientIP=--][ServerIP=--][SubjectID=CN=PKI Administrator,E=example@testdomain.com,OU=rhcs94-CA-cfu_rsa-nocp11,O=Example-rhcs94-CA_cfu-rsa][Outcome=Success][Info=serverAlertReceived: CLOSE_NOTIFY] access session terminated
0.https-jsse-nio-18443-exec-16 - [29/Sep/2021:21:11:34 EDT] [14] [6] [AuditEvent=ACCESS_SESSION_TERMINATED][ClientIP=--][ServerIP=--][SubjectID=CN=PKI Administrator,E=example@testdomain.com,OU=rhcs94-CA-cfu_rsa-nocp11,O=Example-rhcs94-CA_cfu-rsa][Outcome=Success][Info=serverAlertSent: CLOSE_NOTIFY] access session terminated"

This was a change made by @jmagne to fix bug: 2001169 for RHCS 10.2.1, Originally he only made the change on the `v10.11` branch but it needs to go to `master` and `v10.12` as well. 

Tested locally by building and installing JSS, tomcatjss, and pki and observed the AUDIT logs:
```
0.https-jsse-nio-8443-exec-15 - [05/Nov/2021:15:31:17 GMT] [14] [6] [AuditEvent=ACCESS_SESSION_ESTABLISH][ClientIP=--][ServerIP=--][SubjectID=--][Outcome=Success] access session establish success
0.https-jsse-nio-8443-exec-16 - [05/Nov/2021:15:31:17 GMT] [14] [6] [AuditEvent=ACCESS_SESSION_TERMINATED][ClientIP=--][ServerIP=--][SubjectID=--][Outcome=Success][Info=serverAlertReceived: CLOSE_NOTIFY] access session terminated
```

This PR is dependent on JSS PR: https://github.com/dogtagpki/jss/pull/822

### CODE REVIEW

File: base/tps/src/main/java/org/dogtagpki/server/tps/processor/TPSProcessor.java [hunk 1]
```diff
@@ -2387,6 +2387,7 @@ need to reach out to the Registration DB (authid)
             try {
                 tps.tdb.tdbAddTokenEntry(tokenRecord, TokenStatus.UNFORMATTED);
                 tps.tdb.tdbActivity(ActivityDatabase.OP_ADD, tokenRecord, session.getIpAddress(), logMsg, "success");
+                fillTokenRecordDefaultPolicy(tokenRecord);
```
Comment (edewata) [inline base/tps/src/main/java/org/dogtagpki/server/tps/processor/TPSProcessor.java:2390]: @jmagne Is this needed to fix this bug? If not maybe we should put it in a separate commit to avoid confusions (assuming it's needed for the `master` branch).

Comment (jmagne) [reply base/tps/src/main/java/org/dogtagpki/server/tps/processor/TPSProcessor.java:2390]: edewata, under the crush we just put that tps related fix in this blob. I think it should be fine for master. This has to do with tps pin reset.

Comment (jmagne) [reply base/tps/src/main/java/org/dogtagpki/server/tps/processor/TPSProcessor.java:2390]: Also edewata, this is just a missing piece of another commit that covers this pin reset issue. So that issue is represented already.

Comment (edewata) [reply base/tps/src/main/java/org/dogtagpki/server/tps/processor/TPSProcessor.java:2390]: Thanks for the clarification! This is fine.

**Review by edewata — APPROVED** [review]
There seems to be an empty commit added to this PR. I'd suggest squashing it before merging. Other than that, it LGTM.

File: base/server/src/main/java/com/netscape/cmscore/apps/CMSEngine.java [hunk 2]
```diff
@@ -1128,16 +1130,19 @@ public void start() throws Exception {
...
         isStarted = true;
 
         mStartupTime = System.currentTimeMillis();
 
         logger.info(name + " engine started");
+        // Register TomcatJSS socket listener
+        TomcatJSS tomcatJss = TomcatJSS.getInstance();
+       if(serverSocketListener == null) {
+           serverSocketListener = new PKIServerSocketListener();
+        }
```
Comment (ckelleyRH) [inline base/server/src/main/java/com/netscape/cmscore/apps/CMSEngine.java:1143]: Funky whitespace here?

**Comment by c-dorney** [comment]
> There seems to be an empty commit added to this PR. I'd suggest squashing it before merging. Other than that, it LGTM.

@edewata 
Will do!

**Comment by c-dorney** [comment]
> You've still got the empty commit.

I do, I was planning on doing the "Squash and merge" option when merging the PR. I can squash prior to merging if that is preferred.

**Comment by ckelleyRH** [comment]
> > You've still got the empty commit.
> 
> @ckelleyRH I do, I was planning on doing the "Squash and merge" option when merging the PR. I can squash prior to merging if that is preferred.
> 
> edit: Just re-reading Endi's suggestion sweat_smile

Squash and merge is fine, I was just reminding you in case you missed it as you made a revision and you didn't squash it as part of that :+1:

**Review by ckelleyRH — APPROVED** [review]
